### ISSUE TITLE
If a child-prefix is already allocated, we either need to return a warning to the user or reject the join

### ISSUE THREAD

**Issue by nerdalert** [nexodus-io/nexodus#1234]
If a child-prefix is already allocated, we either need to return a warning to the user or reject the join. Currently it's the worst of both, it let's the user join, but only throws a cryptic DB error log on the API server of `/device.go:433","error":"duplicated key not allowed","rows":0,"elapsed":3.436688}`.

Personally, I think the join should be rejected if the network is already being advertised from another node. wdyt?

I added a bug label, but it isn't necessarily a bug but a bad UX currently. Replicate the scenario by advertising the same prefix from two different devices.

Ty!

**Comment by russellb** [comment]
@nerdalert 

> Personally, I think the join should be rejected if the network is already being advertised from another node. wdyt?

Yes, if we can't handle multiple nodes advertising the same prefix, I agree that joining should be rejected.

I also think we should have an enhancement in our backlog so that we CAN handle that properly, but I consider it a new feature.

**Comment by nerdalert** [comment]
Looked into the scenario of the same prefix getting advertised from different devices. I think I even labbed it up. If a node has the same prefix in the allowed list to multiple endpoints, WG won't disallow it but it will break traffic for that prefix to either device. Seems like the obvious work around would be another interface.

Eventually its probably a driver specific capability since that ECMP type limitation is wireguard (possibly ipsec as well) specific.

**Comment by nerdalert** [comment]
After discussing this topic this morning * I think * we landed on:

1. Allow overlapping with better logging/warning in the case of an overlap. 
2. In the long run, the UI will handle overlapping by giving the user the option to allow the advertisement with a switch in the UI.
